x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferModelAction.java

```diff
@@ -179,6 +283,14 @@ public static Builder builder() {
             return new Builder();
         }
 
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field("inference_results", inferenceResults.stream().map(InferenceResults::asMap).collect(Collectors.toList()));
```

Comment:
Alternatively `inferenceResults` which implement `ToXContentFragment` could be mapped to `ToXContentObject`

```
        static ToXContentObject toXContentObject(ToXContentFragment fragment) {
            return (builder, params) -> {
                builder.startObject();
                fragment.toXContent(builder, params);
                builder.endObject();
                return builder;
            };
        }
```

and 

```
builder.field("inference_results", inferenceResults.stream().map(Response::toXContentObject).collect(Collectors.toList()));
```